multimodal/src/autogluon/multimodal/learners/base.py

```diff
@@ -0,0 +1,2472 @@
+            log_filter = LogFilter(blacklist_msgs)
+
+            with apply_log_filter(log_filter):
+                trainer = pl.Trainer(
+                    accelerator="gpu" if num_gpus > 0 else "auto",
+                    devices=num_gpus if num_gpus > 0 else "auto",
+                    num_nodes=self._config.env.num_nodes,
+                    precision=precision,
+                    strategy=strategy,
+                    benchmark=False,
+                    enable_progress_bar=False if barebones else self._enable_progress_bar,
+                    deterministic=self._config.env.deterministic,
+                    max_epochs=-1,  # Add max_epochs to disable warning
+                    logger=False,
+                    callbacks=callbacks,
+                    barebones=barebones,
+                )
+
+        return trainer
+
+    @staticmethod
+    def run_trainer(
+        trainer,
+        task,
+        datamodule,
+        ckpt_path=None,
+        resume=None,
+        pred_writer=None,
+        is_train=True,
+    ):
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore",
+                ".*does not have many workers which may be a bottleneck. "
+                "Consider increasing the value of the `num_workers` argument` "
+                ".* in the `DataLoader` init to improve performance.*",
+            )
+            warnings.filterwarnings("ignore", "Checkpoint directory .* exists and is not empty.")
+            if is_train:
+                trainer.fit(
+                    task,
+                    datamodule=datamodule,
+                    ckpt_path=ckpt_path if resume else None,  # this is to resume training that was broken accidentally
+                )
+            else:
+                outputs = trainer.predict(
+                    task,
+                    datamodule=datamodule,
+                    return_predictions=pred_writer is None,
+                )
+                return outputs
+
+    def on_fit_per_run_start(self, seed, save_path):
+        pl.seed_everything(seed, workers=True)
+        # TODO(?) We should have a separate "_pre_training_event()" for logging messages.
+        logger.info(get_fit_start_message(save_path, self._validation_metric_name))
+
+    def on_fit_per_run_end(
+        self,
+        trainer,
+        config,
+        model,
+        df_preprocessor,
+        data_processors,
+        save_path,
+        standalone,
+    ):
+        self.clean_trainer_processes(trainer=trainer, is_train=True)
+        self.save(
+            path=save_path,
+            standalone=standalone,
+            config=config,
+            model=model,
+            df_preprocessor=df_preprocessor,
+            data_processors=data_processors,
+            fit_called=True,  # fit is called on one run.
+            save_model=False,  # The final model will be saved in top_k_average
+        )
+
+    def fit_per_run(
```

Comment:
So this equal to previous `_fit()`?

Reply:
Right. The new name is more intuitive. HPO needs to call this function many times.